Revise 'Kubernetes Components' page

KhushPatibandha opened the pull request:
Worked on Issue #39562, as per the description suggested.

nate-double-u commented on content/en/docs/reference/glossary/kube-scheduler.md:
```diff
@@ -10,10 +10,9 @@ aka:
-{{< glossary_tooltip term_id="pod" text="Pods" >}} with no assigned
-{{< glossary_tooltip term_id="node" text="node">}}, and selects a node for them
-to run on.
+Control plane component that watches 
+for newly created {{< glossary_tooltip term_id="pod" text="Pods" >}} 
+that have not yet been assigned to any {{< glossary_tooltip term_id="node" text="node">}}
```
```suggestion
A control plane component that watches 
for new, unassigned {{< glossary_tooltip term_id="pod" text="Pods" >}} 
and chooses a {{< glossary_tooltip term_id="node" text="node">}} for them.
```

KhushPatibandha replied:
suggested changes done.

nate-double-u reviewed (commented):
Hi @KhushPatibandha, thanks for this. We may be able to simplify further, and listing out the scheduling decision factors in a bulleted list may help with readability.

KhushPatibandha commented:
Sure I'll look into it.

KhushPatibandha commented:
@nate-double-u So I was only suppose to add bullet points as you said but I also added an explanation for all the factors taken into account for scheduling decisions, I hope that fine. We can cut those out if you think that's not the correct way to put it, but nonetheless I have added bullet points. Do let me know what my next steps should be if the current files changes are not up to the standards.

KhushPatibandha commented:
@nate-double-u what's the update on this one?

a-mccarthy commented on content/en/docs/reference/glossary/kube-scheduler.md:
```diff
@@ -10,14 +10,52 @@ aka:
+- inter-workload interference
+- deadlines
+
+Individual and collective resource requirements: The CPU and memory requirements 
```
@KhushPatibandha I like the bulleted list, but I think the rest might be too much information for the glossary section. I'd suggest we add a link to a page where we talked about these in more detail instead. Maybe this page, https://kubernetes.io/docs/concepts/scheduling-eviction/kube-scheduler/ 

And if we dont have this information written out in the docs yet, and you think it should still be added, please consider creating a new PR adding the content to kube-scheduler page

nate-double-u replied:
@KhushPatibandha, I'm sorry I lost the thread on this, but agree with @a-mccarthy's feedback here.

a-mccarthy commented:
Hi @KhushPatibandha thanks for working on this, sorry for the delay in reviewing it. I just left a comment with some suggestions. If you still have time to work on this PR, I'd be interested in hearing your thoughts.

a-mccarthy commented on content/en/docs/reference/glossary/kube-scheduler.md:
```diff
@@ -10,14 +10,52 @@ aka:
-{{< glossary_tooltip term_id="pod" text="Pods" >}} with no assigned
-{{< glossary_tooltip term_id="node" text="node">}}, and selects a node for them
-to run on.
+A control plane component that watches 
```
If we are improving the longer versions of the glossary, I think it also makes sense to update the `short_description` in the front matter at the top of the file to use the same wording.

KhushPatibandha commented:
hey @a-mccarthy thanks for the feedback. I have updated the `short_description` and removed the detailed explanation of those bullet points. Do lemme know if adding a detailed explanation of those bullet point add any value to the docs, if yes, i would love to contribute to it.

sftim commented:
/retitle Revise 'Kubernetes Components' page

sftim commented:
/hold

We prefer not to mention issues and PRs in commit messages, so https://github.com/kubernetes/website/pull/40129/commits/ae24f03b2ce3bee7ecd76baedcf7d0cef75fa38a is a problem here.

@KhushPatibandha would you like any advice on fixing that?

OK to unhold once https://github.com/kubernetes/website/pull/40129/commits looks clean.

divya-mohan0209 commented:
@KhushPatibandha : Gentle reminder on the PR.

divya-mohan0209 commented:
Hello @KhushPatibandha , we haven't received any further commits/comments around the suggestions made. On account of this, we shall be closing this PR. When you have the bandwidth, please feel free to reopen and resume work on it.
/close
